clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -157,12 +200,26 @@ object UncommittedGarbageCollector {
       firstSlice: String,
       startTime: java.time.Instant,
       success: Boolean,
-      removed: DataFrame
+      expiredAddresses: DataFrame
   ): Unit = {
-    val reportDst = s"${storageNamespace}_lakefs/retention/gc/uncommitted/$runID"
-    writeJsonSummary(reportDst, runID, firstSlice, startTime, success, removed.count())
+    val reportDst = reportPath(storageNamespace, runID)
+    writeJsonSummary(reportDst, runID, firstSlice, startTime, success, expiredAddresses.count())
+    expiredAddresses.show()
+    expiredAddresses.write.parquet(s"${reportDst}/deleted")
+  }
+
+  private def reportPath(storageNamespace: String, runID: String): String = {
+    s"${storageNamespace}_lakefs/retention/gc/uncommitted/${runID}"
+  }
 
-    removed.write.parquet(s"$reportDst/deleted")
+  private def readMarkedAddresses(storageNamespace: String, markID: String): DataFrame = {
+    val path = reportPath(storageNamespace, markID)
+    val markedRunSummary = spark.read.json(s"${path}/summary.json")
+    if (markedRunSummary.select("success").first() == false) {
```

Comment:
We should throw an exception here. We ignore the failed run status and the user assumes his sweep run completed successfully

Reply:
It is caught in the try-catch of main